### PR TITLE
[SPARK-49636][SQL] Remove the ANSI config suggestion in INVALID_ARRAY_INDEX and INVALID_ARRAY_INDEX_IN_ELEMENT_AT

### PR DESCRIPTION
**Pull request by mihailomilosevic2001**
### What changes were proposed in this pull request?
Removal of ansi turn off suggestion for INVALID_ARRAY_INDEX and INVALID_ARRAY_INDEX_IN_ELEMENT_AT.

### Why are the changes needed?
Now that in Spark 4.0.0 we have moved to ANSI mode on by default, we want to keep suggestions of this kind to the minimum. The good thing is that both errors have other suggestions which are sufficient.

### Does this PR introduce _any_ user-facing change?
Yes, error message has changed.

### How was this patch tested?
Existing tests.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Review by srielau — APPROVED** [review]

File: docs/sql-ref-ansi-compliance.md [hunk 1]
```diff
@@ -384,6 +384,8 @@ When ANSI mode is on, it throws exceptions for invalid operations. You can use t
   - `try_make_timestamp_ltz`: identical to the function `make_timestamp_ltz`, except that it returns `NULL` result instead of throwing an exception on error.
   - `try_make_timestamp_ntz`: identical to the function `make_timestamp_ntz`, except that it returns `NULL` result instead of throwing an exception on error.
   - `try_make_interval`: identical to the function `make_interval`, except that it returns `NULL` result instead of throwing an exception on invalid interval.
+  - `try_elt`: identical to the function `elt`, except that it returns `NULL` result instead of throwing an exception on index out of bounds.
```
Comment (MaxGekk) [inline docs/sql-ref-ansi-compliance.md:387]: where is `try_elt` defined?

File: docs/sql-ref-ansi-compliance.md [hunk 2]
```diff
@@ -384,6 +384,8 @@ When ANSI mode is on, it throws exceptions for invalid operations. You can use t
...
+  - `try_get_array_item`
```
Comment (MaxGekk) [inline docs/sql-ref-ansi-compliance.md:388]: The same question as above + comment.

File: docs/sql-ref-ansi-compliance.md (hunk 1 above)
Comment (mihailom-db) [reply docs/sql-ref-ansi-compliance.md]: You are right, I was thinking about whether we need this one, but it turns out we can use `get` + `explode` as a workaround.

File: docs/sql-ref-ansi-compliance.md (hunk 2 above)
Comment (mihailom-db) [reply docs/sql-ref-ansi-compliance.md]: Ditto.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @mihailom-db and @srielau for review.

**Pull request closed without merging**